.github/workflows/build.yml

```diff
@@ -0,0 +1,53 @@
+name: Build
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  packages: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@2b82ce82d56a2a04d2637cd93a637ae1b359c0a7 # v2.2.0
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@4c0219f9ac95b02789c1075625400b2acbff50b1 # v2.9.1
+
+      - name: Login to GHCR
+        if: github.ref == 'refs/heads/main'
+        uses: docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc # v2.2.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@818d4b7b91585d195f67373fd9cb0332e31a7175 # v4.6.0
+        with:
+          images: |
+            ghcr.io/${{ github.repository }}
+          flavor: |
+            latest=false
+          tags: |
+            type=semver,pattern={{version}}
```

Comment:
where does it get the version from?

Reply:
This is documented in the action at https://github.com/docker/metadata-action#typesemver.

The `type=semver` is triggered on git tags, with the `pattern={{version}}` being the tag with a potential leading `v` removed.

Reply:
For now, this of course won't do anything, we'd get the following for now on merge to main: `ghcr.io/renovatebot/renovate-approve-bot:main`.

Tags would then start to be a thing with the introduction of versioning, which I think will be done with semantic-release.
If you prefer to switch that around and first introduce versioning and then the image build, we can do that, too.